### PR TITLE
Update command for eco reset

### PR DESCRIPTION
**Pull request by Destitution-SC**
Fixes issue #3251 @pop4959

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/commands/Commandeco.java [hunk 1]
```diff
@@ -16,8 +16,8 @@
 
 import static com.earth2me.essentials.I18n.tl;
 
-
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandeco.java]: Undo this diff to keep consistent style with other ess code

File: Essentials/src/com/earth2me/essentials/commands/Commandeco.java [hunk 2]
```diff
@@ -16,8 +16,8 @@
...
 public class Commandeco extends EssentialsLoopCommand {
+
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandeco.java]: =

File: Essentials/src/com/earth2me/essentials/commands/Commandeco.java [hunk 3]
```diff
@@ -103,6 +106,14 @@ private void set(BigDecimal amount, final User player, final CommandSource sende
         }
     }
 
+    private void reset(final User player, final CommandSource sender) throws MaxMoneyException {
+        BigDecimal startingBalance = ess.getSettings().getStartingBalance();
+        player.setMoney(startingBalance, UserBalanceUpdateEvent.Cause.COMMAND_ECO);
+        player.sendMessage(tl("resetBalSelf", startingBalance.toString()));
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandeco.java]: Should probably be using NumberUtil::displayCurrency here

File: Essentials/src/com/earth2me/essentials/commands/Commandeco.java [hunk 4]
```diff
@@ -103,6 +106,14 @@ private void set(BigDecimal amount, final User player, final CommandSource sende
...
+        if(sender != null) {
+            sender.sendMessage(tl("resetBalOther", player.getDisplayName(), startingBalance.toString()));
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandeco.java]: =

**Review by pop4959 — CHANGES_REQUESTED** [review]
In addition to the comments, can you show that you have tested it (command(s) used + screenshot(s) are good enough).

**Comment by pop4959** [comment]
Oh also it might be good to change the usage for eco in the plugin.yml file to indicate that the amount argument is optional.

**Comment by Destitution-SC** [comment]
Welp it doesn't work it keeps returining the usage. :(
I can a bit code but im not really experienced. So can anyone else help?

**Comment by imDaniX** [comment]
Can't we just do something like this?
```java
            case RESET:
                amount = ess.getSettings().getStartingBalance();

            case SET:
                set(amount, player, sender);
                break;
```
So we don't have to create new tr strings.
Also, `/eco reset` is a part of `/eco`, so I'm not sure how it supposed to look like because it will change other parts of the command.
Maybe `/<command> <give|take|set|reset> <player> [amount]`, but it's not really good.

File: Essentials/src/plugin.yml [hunk 5]
```diff
@@ -118,7 +118,7 @@ commands:
     aliases: [edisposal,trash,etrash]
   eco:
     description: Manages the server economy.
-    usage: /<command> <give|take|set|reset> <player> <amount>
+    usage: /<command> [give|take|set|reset] [player] <amount>
```
Comment (pop4959) [inline Essentials/src/plugin.yml]: Original was correct, no need for change here. The angle brackets (by common convention) indicate *required* arguments, while square brackets indicate *optional* arguments. In this command, the first two arguments are required, however the third can be optional. So you actually want to flip this here.

**Review by pop4959 — CHANGES_REQUESTED** [review]
Sorry for the long delay in reviewing your PR, I've been more busy lately. Someone recently suggested an implementation for this fix which may be simpler, which is to simply set the `amount` to the configured starting balance in the reset command, rather than writing a lot of extra logic (especially since a lot of it ends up being similar, and involves more translation). Maybe give it a try.

As for your question, it is probably returning the usage because the command still expects  (as is checking) that all 3 arguments are passed in. You might have to add an exception to that for reset.

If you need help answering any other questions let me know.

**Comment by Destitution-SC** [comment]
Screenshot_1 I'm pretty sure these are the errors returning the usage but no idea on how to fix that error with try and catch thing.

**Comment by pop4959** [comment]
Hey @Destitution-SC, thanks for all of the effort you put in here. I decided I'd help you out and take a look at the issue myself, and it looks like the fix is actually quite easy. Check out #3340. So, the solution is just to short circuit evaluate `isPercent` in a similar manner to `amount`, because the problem is with accessing `args[2]` when there is no third argument present (which is the case when reset is run). 😄

**Pull request closed without merging**